### ISSUE TITLE
Dafny crash in Translator.FuelSetting.GetFunctionFuel

### ISSUE THREAD

**Issue by jaylorch** [Microsoft/dafny#1]
Dafny crashes in Translator.FuelSetting.GetFunctionFuel when verifying the code below:

```
datatype T = T(n:int)

function ToInt(t:T) : int
    ensures ToInt(t) == int(t.n);
{
    int(t.n)
}

method Test()
{
    assert exists p:int :: exists t:T :: ToInt(t) > 0;
}
```

**Comment by qunyanm** [comment]
Fixed

**Comment by cpitclaudel** [comment]
Neat :) First fixed issue on Github :+1: :tada: ! Btw @qunyanm, a neat feature of github is that if you write "Fixes #1" or "Closes #1" in the commit message, it automatically closes the corresponding ticket with a link to the commit (see https://github.com/cpitclaudel/company-coq/issues/124 for an example)

**Comment by qunyanm** [comment]
Oh, I didn't know the feature. It will be handy. Thanks for letting me know.
